File: core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.java
package com.larryhsiao.nyx.core.jots;

import com.silverhetch.clotho.Action;
import com.silverhetch.clotho.Source;

import java.sql.Connection;
import java.sql.PreparedStatement;

/**
 * Action to remove a Jot
 */
public class JotRemoval implements Action {
    private final Source<Connection> db;
    private final long id;

    public JotRemoval(Source<Connection> db, long id) {
        this.db = db;
        this.id = id;
    }

    @Override
    public void fire() {
        try {
            final PreparedStatement removeTag = db.value().prepareStatement(
                // language=H2
                "DELETE FROM TAG_JOT " +
                    "WHERE JOT_ID=?"
            );
            removeTag.setLong(1, id);
            removeTag.executeUpdate();
            removeTag.close();

            final PreparedStatement removeJot = db.value().prepareStatement(
                // language=H2
                "DELETE FROM JOTS " +
                    "WHERE ID=?"
            );
            removeJot.setLong(1, id);
            removeJot.executeUpdate();
            removeJot.close();
        } catch (Exception e) {
            throw new IllegalArgumentException(e);
        }
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/JotRemoval.kt
package com.larryhsiao.nyx.core.jots

import com.larryhsiao.clotho.Action
import com.larryhsiao.clotho.Source
import java.sql.Connection

/**
 * Action to remove a Jot
 */
class JotRemoval(private val db: Source<Connection>, private val id: Long) : Action {
    override fun fire() {
        try {
            val removeTag = db.value().prepareStatement( // language=H2
                "UPDATE TAG_JOT " +
                    "SET DELETE=1 , VERSION = VERSION + 1 " +
                    "WHERE JOT_ID=?"
            )
            removeTag.setLong(1, id)
            removeTag.executeUpdate()
            removeTag.close()
            val removeJot = db.value().prepareStatement( // language=H2
                "UPDATE JOTS " +
                    "SET DELETE=1, VERSION = VERSION + 1 " +
                    "WHERE ID=?"
            )
            removeJot.setLong(1, id)
            removeJot.executeUpdate()
            removeJot.close()
        } catch (e: Exception) {
            throw IllegalArgumentException(e)
        }
    }
}